src/OSPSuite.R/Services/ContainerTask.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using OSPSuite.Assets;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Core.Extensions;
+using OSPSuite.Utility.Exceptions;
+using OSPSuite.Utility.Extensions;
+using static OSPSuite.Core.Domain.Constants;
+
+namespace OSPSuite.R.Services
+{
+   public interface IContainerTask
+   {
+      IParameter[] AllParametersMatching(IModelCoreSimulation simulation, params string[] path);
+      IContainer[] AllContainersMatching(IModelCoreSimulation simulation, params string[] path);
+
+      IParameter[] AllParametersMatching(IContainer container, params string[] path);
+      IContainer[] AllContainersMatching(IContainer container, params string[] path);
+   }
+
+   public class ContainerTask : IContainerTask
+   {
+      private readonly IEntityPathResolver _entityPathResolver;
+      private static readonly string ALL_BUT_PATH_DELIMITER = $"[^{ObjectPath.PATH_DELIMITER}]*";
+
+      public ContainerTask(IEntityPathResolver entityPathResolver)
+      {
+         _entityPathResolver = entityPathResolver;
+      }
+
+      public IParameter[] AllParametersMatching(IModelCoreSimulation simulation, params string[] path) =>
+         AllParametersMatching(simulation?.Model?.Root, path);
+
+      public IContainer[] AllContainersMatching(IModelCoreSimulation simulation, params string[] path) =>
+         AllContainersMatching(simulation?.Model?.Root, path);
+
+      public IContainer[] AllContainersMatching(IContainer container, params string[] path) =>
+         allEntitiesMatching<IContainer>(container, path);
+
+      public IParameter[] AllParametersMatching(IContainer container, params string[] path) =>
+         allEntitiesMatching<IParameter>(container, path);
+
+      private T[] allEntitiesMatching<T>(IContainer container, string[] path) where T : class, IEntity
+      {
+         if (path == null || path.Length == 0)
+            return Array.Empty<T>();
+
+         validate(path);
+         var pathAsString = path.ToPathString();
+
+         // no wild cards => it's a single path and do not need to inspect 
+         if (!pathAsString.Contains(WILD_CARD))
+         {
+            var entity = container.EntityAt<T>(path);
+            return entity == null ? Array.Empty<T>() : new[] {entity};
+         }
+
+         var regex = new Regex(createSearchPattern(path), RegexOptions.IgnoreCase);
+         var parentContainerPath = $"{_entityPathResolver.FullPathFor(container)}{ObjectPath.PATH_DELIMITER}";
+
+         return container.GetAllChildren<T>(x => pathMatches(regex, parentContainerPath, x)).ToArray();
+      }
+
+      private void validate(string[] path)
```

Comment:
@Yuri05 Ok I am now checking that ** is not used as part of path.,
Message  is also quite explicit